include/mqtt/client.hpp

```diff
@@ -254,6 +254,14 @@ class client : public endpoint<std::mutex, std::lock_guard, PacketIdBytes> {
         client_id_ = force_move(id);
     }
 
+    /**
+     * @brief Get the client id.
+     * @return The client id of this client.
+     */
+    const std::string& get_client_id (void) const {
```

Comment:
```suggestion
    std::string const& get_client_id() const {
```

I am sorry for not being able to comment all at once.

* C++ use `()` for empty parameter instead of `(void)`
* mqtt_cpp uses east const 
   * See https://github.com/redboltz/mqtt_cpp/wiki/Coding-Rules
* mqtt_cpp doesn't put white space between function name and `(`

Could you update the PR. Then I will merge.